src/util/instanceMigration.tsx

```diff
@@ -0,0 +1,134 @@
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+import { useEventQueue } from "context/eventQueue";
+import { useInstanceLoading } from "context/instanceLoading";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { queryKeys } from "./queryKeys";
+import { fetchStoragePools } from "api/storage-pools";
+import ItemName from "components/ItemName";
+import { migrateInstance } from "api/instances";
+import { LxdInstance } from "types/instance";
+import { fetchClusterMembers } from "api/cluster";
+import { ReactNode, useState } from "react";
+import { useSettings } from "context/useSettings";
+import { isClusteredServer } from "./settings";
+import { capitalizeFirstLetter } from "./helpers";
+
+export type MigrationType = "cluster member" | "root storage pool" | "";
+
+type Props = {
+  instance: LxdInstance;
+  close: () => void;
+};
+
+export const useInstanceMigration = (props: Props) => {
+  const { instance, close } = props;
```

Comment:
```suggestion
export const useInstanceMigration = ({ instance, close, type, target }: Props) => {
```
We tend to destructure the props in the header everywhere else in the repo.